protostar-rust/src/lib.rs

```diff
@@ -137,6 +150,28 @@ fn run_tests(tests: TestsFromFile, tests_stats: &mut TestsStats) -> Result<()> {
     Ok(())
 }
 
+fn filter_tests_by_name(
+    test_name_filter: Option<&str>,
+    test_configs: Vec<TestConfig>,
+) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    if let Some(test_name_filter) = test_name_filter {
+        for test in test_configs {
+            let name = test
+                .name
+                .rsplit("::")
+                .next()
+                .context("Failed to get test")?;
+            if name.contains(test_name_filter) {
```

Comment:
I don't think it is a good design choice especially since we match only against the name, not the full path. There might be cases where it is not possible to run test A without running test B. Keep in mind that Protostar tests might need a lot of time to execute so it's a major inconvenience imo

Reply:
We can extend the logic so it also matches by crate etc. I'd avoid requiring to explicitly pass the whole test name, it's even more impractical in my opinion.

Reply:
Discussed offline